samples/DLLMethodCaller/FortuneTeller.cs

```diff
@@ -7,15 +7,23 @@ namespace DLLMethodCaller
 {
     class FortuneTeller
     {
+        private int total;
+        private int fortuneTellerFee = 1;
         public void AcceptMoney(int amount)
         {
-            // TODO
+            total += amount;
         }
 
         public string PredictFuture(DateTime birthday, bool isMarried)
         {
-            // TODO
-            return "";
+            total -= fortuneTellerFee;
+            if (total ==0 || isMarried)
+                return "You're broke!";
+            if (total < 0 )
+                return "You're in debt!";
+
+            Random random = new Random(int.Parse(birthday.ToString("MMddHHmmss")));
+            return $"You're {(total<100?"not so":"")} rich. You're worth {random.Next(0, total)} today!";
```

Comment:
Let's clarify the logic. If I understand your idea right you want to:
1) make a funny future prediction (random.Next); 
2) check if that number is < 100: and say "You're not so rich in the future."
3) if the number is > 100 and say "You're rich. You're worth $X!"

To simplify it I propose to extract a new variable here (`int prediction = random.Next...`), and extract an "if" condition such as you do above.

P.S. I'm not an expert in future predictions :) but I would not base it on the "total", I mean the whole point is that you could give 1$, but you might get a promise to get 1M$ and be rich :)

Reply:
Something like that?

       int prediction = random.Next(0, total);
       string answer;
       if (total < 100)
            {
                answer = $"You're not rich but also not broke. That's a good thing. You're worth {prediction} today!";
            }
            else
            {
                answer = $"You're rich! Well done! You're worth {prediction} today!";
            }